pkg/flags/flag.go

```diff
@@ -123,6 +123,10 @@ func verifyEnv(prefix string, usedEnvKey, alreadySet map[string]bool) {
 			plog.Warningf("recognized environment variable %s, but unused: shadowed by corresponding flag", kv[0])
 			continue
 		}
+		// unsupported architecture only configured via environment variable
+		if env == "ETCD_UNSUPPORTED_ARCH" {
```

Comment:
the flag pkg should not understand etcd specific thing. can we do something similar to usedEnvKey?

Reply:
Problem is that `usedEnvKey` is only populated when there is a matching *flag*. We don't have flag `unsupported-flag` in the flag set, so it warns.

Reply:
then how about cleaning `ETCD_UNSUPPORTED_ARCH` before we calling into this pkg in etcdmain or somewhere.